From: Mike Snitzer <snitzer@redhat.com>
Date: Wed, 17 Feb 2010 00:31:37 -0500
Subject: [dm] stripe: avoid divide by zero with invalid stripe count
Message-id: <1266366701-1575-2-git-send-email-snitzer@redhat.com>
Patchwork-id: 23302
O-Subject: [RHEL6 PATCH 1/5] dm stripe: avoid divide by zero with invalid stripe
	count
Bugzilla: 565890
RH-Acked-by: Alasdair G Kergon <agk@redhat.com>

Bugzilla #565890

commit 781248c1b50c776a9ef4be1130f84ced1cba42fe
Author: Nikanth Karthikesan <knikanth@suse.de>
Date:   Tue Feb 16 18:42:47 2010 +0000

    dm stripe: avoid divide by zero with invalid stripe count

    If a table containing zero as stripe count is passed into stripe_ctr
    the code attempts to divide by zero.

    This patch changes DM_TABLE_LOAD to return -EINVAL if the stripe count
    is zero.

    We now get the following error messages:
      device-mapper: table: 253:0: striped: Invalid stripe count
      device-mapper: ioctl: error adding target to table

    Signed-off-by: Nikanth Karthikesan <knikanth@suse.de>
    Cc: stable@kernel.org
    Signed-off-by: Alasdair G Kergon <agk@redhat.com>

Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>

diff --git a/drivers/md/dm-stripe.c b/drivers/md/dm-stripe.c
index e0efc1a..bd58703 100644
--- a/drivers/md/dm-stripe.c
+++ b/drivers/md/dm-stripe.c
@@ -110,7 +110,7 @@ static int stripe_ctr(struct dm_target *ti, unsigned int argc, char **argv)
 	}
 
 	stripes = simple_strtoul(argv[0], &end, 10);
-	if (*end) {
+	if (!stripes || *end) {
 		ti->error = "Invalid stripe count";
 		return -EINVAL;
 	}
